### PR TITLE
Add Enum support to binding generator

### PR DESCRIPTION
**Pull request by Krzysztof-Cieslak**
As discussed in #34

### CODE REVIEW

**Comment by Krzysztof-Cieslak** [comment]
Also adding node.d.ts and typescript.d.ts files and referencing them in generator file - it enables having JS autocomplete, tooltips etc. in VS Code

File: tools/typescript2fable.js [hunk 1]
```diff
@@ -310,9 +335,12 @@ function printModule(prefix) {
     return function(mod) {
         var template = prefix + templates.module
             .replace("[NAME]", escapeKeyword(mod.name));
-            
-        template = append(template, mod.interfaces.map(
-            printInterface(prefix + "    ", mod.name)).join("\n\n"));
+
+        var interfaces = mod.interfaces && mod.interfaces.length > 0
+                            ? mod.interfaces.map(printInterface(prefix + "    ", mod.name)).join("\n\n")
+                            : "";
```
Comment (alfonsogarciacaro) [inline tools/typescript2fable.js]: What are the edge cases this is trying to solve? `mod.interfaces` should never be `undefined` and executing `map` + `join` on an empty array should give an empty string so the check `mod.interfaces.length > 0` shouldn't be necessary

File: tools/typescript2fable.js [hunk 2]
```diff
@@ -497,7 +536,8 @@ function getModule(name) {
       interfaces: [],
       properties: [],
       methods: [],
-      modules: []
+      modules: [],
+      enums: []
```
Comment (alfonsogarciacaro) [inline tools/typescript2fable.js]: We need to add a check for `enums` length in https://github.com/fsprojects/Fable/blob/master/tools/typescript2fable.js#L577. Or maybe we can do something generic like checking at least one array property (`Array.isArray(mod[k])`) satisfies `length > 0`.

**Comment by alfonsogarciacaro** [comment]
Great work, thank you! I added a couple of comments, can you please check? Also, having typescript definitions is a great idea (I was too lazy to setting them up) but as the files are very big I think it'd be cleaner if we add them to npm dependencies instead.

File: tools/typescript2fable.js (hunk 1 above)
Comment (Krzysztof-Cieslak) [inline tools/typescript2fable.js]: Not sure to be fair, when I was playing with this code, I've expected exception here... but that could be result of me messing something in other parts of code.

**Pull request merged**